build/build.py

```diff
@@ -1053,6 +1053,11 @@ def configureCompilerOptions(self):
                   '_LARGEFILE_SOURCE WIN32 _USE_MATH_DEFINES NOMINMAX WIN32_LEAN_AND_MEAN'.split()
         flags = '/UUNICODE /U_UNICODE /EHs /GR'.split()
 
+        #If building with cpp17 add flags/defines to enable auto_ptr
+        if Options.options.enablecpp17:
+            defines.append('_HAS_AUTO_PTR_ETC=1')
+            flags.append('/std:c++17')
+
         self.env.append_value('DEFINES', defines)
```

Comment:
`std::auto_ptr` shouldn't be needed in C++17, use `mem::auto_ptr` instead.

Reply:
@JDanielSmith, can we just have the code emit some friendly advice about how to update to newer pointer types here, while still allowing the extra `_HAS_AUTO_PTR_ETC=1` ? 

I could see some older codebases preferring that (which I'm guessing is the driver for this change), since without this change, as soon as you try to play around w/ c++17, you get smacked in the face with build failures.

Reply:
C++17 has **removed** `std::auto_ptr`, it has been deprecated since C++11.  If you want to "play around" w/C++17, **change your code**; `mem::auto_ptr` can help as it is still `std::auto_ptr` pre C++17.

All of **coda-oss**, **nitro** and **six-library** are "clean" in this regard; no special compiler flag needed to bring `std::auto_ptr` into C++17.  If *projects* want to do that, nothing stopping them; but not in **coda-oss**.

Reply:
Okay I will remove that, but I will point out that std::auto_ptr is being enabled in the gcc flags with -Wno-deprecated-declarations, so if it is a problem to have removed features enabled, that flag should be removed as well.

Reply:
Personally, I would have done all of this C++17 stuff in the **main** branch (not **master**) which was already C++17-enabled and does not support anything earlier.

Reply:
The main branch has that compiler flag as well.